src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -348,6 +406,37 @@ private void EnsureSignature()
             }
         }
 
+        [MemberNotNull("_lazyBackingFieldSymbolForFieldKeyword")]
+        private void EnsureBackingFieldIsSynthesized()
+        {
+            // Ensure the binding is done so we guarantee that we have the BackingField set if necessary.
+            if (this is SourcePropertySymbol propertySymbol)
+            {
+                if (_lazyBackingFieldSymbolForFieldKeyword is null && propertySymbol.GetMethod is SourcePropertyAccessorSymbol { ContainsFieldKeyword: true } getMethod)
+                {
+                    noteAccessorBinding();
+                    var binder = getMethod.TryGetBodyBinder();
+                    binder?.BindMethodBody(getMethod.SyntaxNode, BindingDiagnosticBag.Discarded);
+                }
+                if (_lazyBackingFieldSymbolForFieldKeyword is null && propertySymbol.SetMethod is SourcePropertyAccessorSymbol { ContainsFieldKeyword: true } setMethod)
+                {
+                    noteAccessorBinding();
+                    setMethod.TryGetBodyBinder()?.BindMethodBody(setMethod.SyntaxNode, BindingDiagnosticBag.Discarded);
+                }
+            }
+
+            // Avoids re-calculation if we already have calculated it.
+            _lazyBackingFieldSymbolForFieldKeyword ??= new(null);
+
+            void noteAccessorBinding()
+            {
+                if (DeclaringCompilation.TestOnlyCompilationData is AccessorBindingData accessorBindingData)
+                {
+                    accessorBindingData.NumberOfPerformedAccessorBinding++;
```

Comment:
>NumberOfPerformedAccessorBinding [](http://example.com/codeflow?start=40&length=32)

This is not thread safe. I'd change it to a field and use interlocked increment to modify it.